tests/equation_bcs/test_equation_bcs.py

```diff
@@ -339,21 +331,18 @@ def test_EquationBC_mixedpoisson_matfree_fieldsplit(eq_type, mat_type, porder):
     # matfree with fieldsplit pc
 
     solver_parameters = {'mat_type': mat_type,
-                         'ksp_type': 'gmres',
-                         'ksp_atol': 1e-09,
-                         'ksp_rtol': 1e-09,
-                         'ksp_max_it': 200000,
-                         'ksp_divtol': 1e8,
+                         'ksp_type': 'fgmres',
+                         'ksp_atol': 1e-11,
+                         'ksp_max_it': 200,
                          'pc_type': 'fieldsplit',
                          'pc_fieldsplit_type': 'schur',
                          'pc_fieldsplit_schur_fact_type': 'full',
-                         'fieldsplit_0_ksp_type': 'gmres',
-                         'fieldsplit_0_ksp_rtol': 1.e-12,
+                         'fieldsplit_0_ksp_type': 'cg',
```

Comment:
It's concerning I need this, and can't get away with `preonly` for the `ksp_type` here, since the PC should be exact.